packages/components/scripts/wrapper-generator/VueWrapperGenerator.ts

```diff
@@ -0,0 +1,144 @@
+import { AbstractWrapperGenerator } from './AbstractWrapperGenerator';
+import { getComponentMeta } from '@porsche-design-system/shared';
+import type { TagName } from '@porsche-design-system/shared';
+import type { ExtendedProp } from './DataStructureBuilder';
+import { camelCase, pascalCase } from 'change-case';
+
+export class VueWrapperGenerator extends AbstractWrapperGenerator {
+  protected packageDir = 'components-vue';
+  protected projectDir = 'vue-wrapper';
+
+  public getComponentFileName(component: TagName): string {
+    return `${pascalCase(component.replace('p-', ''))}Wrapper.vue`;
+  }
+
+  public generateImports(component: TagName, extendedProps: ExtendedProp[], nonPrimitiveTypes: string[]): string {
+    const hasEventProps = extendedProps.some(({ isEvent }) => isEvent);
+    const hasProps = getComponentMeta(component).props;
```

Comment:
Sticking to `extendedProps` would be cleaner than mixing up two different sources.